### PR TITLE
fix(client): Ensure result extensions are applied after all query extensions

### LINKED ISSUE #20437: Client extensions: result extensions should be applied after query extensions
f you are short-curcuting query call, result extensions would also be skipped. For example:

```ts
prisma.$extends({
    query: {
        $allOperations({ query, args }) {
              if (someCondition) return someMockedResponse
              return query(args)
         }
    }
}).$extends({
   result: {
      someModel: {
         computedField: { ...}
      }
   }
})
```

Imagine they are two separate extensions, not aware of each other and someMockedResponse perfectly mimics DB response for particular query. From the standpoint of mocking extension, it does everything correctly. However, return value of query extension acts as final decider on what gets returned from the query. So, in case mocking condition holds true, someMockedResponse would act as one and final result of the query and result extension will get skipped. So, mocked response won’t contain computedField, unlike normal request.

Result extensions should apply only and always after all query extensions are done and should not be observable in any way from within query extension.

[Internal discussion](https://prisma-company.slack.com/archives/C043X6JBND6/p1690462237248899)

### PR DESCRIPTION
**Pull request by SevInf**
Fix #20437

/integration

### CODE REVIEW

File: packages/client/src/runtime/getPrismaClient.ts [hunk 1]
```diff
@@ -893,6 +900,29 @@ Or read our docs at https://www.prisma.io/docs/concepts/components/prisma-client
       })
     }
 
+    _applyResultExtensions(result: object | null, modelName: string, args: JsArgs) {
```
Comment (millsp) [inline packages/client/src/runtime/getPrismaClient.ts:903]: +1 for moving it to another file near applyQueryExtensions if possible.

**Review by millsp — APPROVED** [review]

File: packages/client/src/runtime/getPrismaClient.ts (hunk 1 above)
Comment (SevInf) [reply packages/client/src/runtime/getPrismaClient.ts:903]: Done.

**Comment by greg-nagy** [comment]
@SevInf I am not sure about the original intention but I assume,  that the result extensions should also work for nested queries as well. At the moment if the model is loaded through a select, then the result extensions are not applied because the model is the "parent" model. 

See my comment for a bit more detail: https://github.com/prisma/docs/issues/5217#issuecomment-1696474897

What would be a good next step here?

**Comment by SevInf** [comment]
@greg-nagy [Result extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result) are intended to work with nested read and writes and we have several tests for that inside of our repo. If that's not the case - please, open a separate issue and share all reproduction steps. 
This PR generally has nothing to do with this. You are incorrect about `applyAllResultExtensions` function being introduced here - it existed ever since result extensions were introduced, just in the difference place under different name.

**Comment by greg-nagy** [comment]
thanks for the response! 

>  are intended to work with nested read and writes ... If that's not the case - please, open a separate issue and share all reproduction steps.

That doesn't seem to be the case — although I am 1 version behind atm. I will create reproduction steps and open an issue when I get there. 

Until then

> You are incorrect about applyAllResultExtensions function being introduced here - it existed ever since result extensions were introduced, just in the difference place under different name.

Might be the case easily. Thanks for the clarification. I just checked the last PR that changed the relevant parts.

**Comment by greg-nagy** [comment]
@SevInf You have just updated your docs with this limitation https://github.com/prisma/docs/pull/5246/files

Does this mean that you are aware of the issue and no need to open a new issue from me?

**Comment by SevInf** [comment]
@greg-nagy, no, I believe that doc change was incorrect, we are working on it.
I'd appreciate if we continued this conversation in a dedicated issue with a reproduction, rather than on unrelated closed PR.

**Pull request merged**